commands/members/remove.js

```diff
@@ -2,24 +2,69 @@
 
 let cli = require('heroku-cli-util')
 let co = require('co')
+let Utils = require('../../lib/utils')
 
 function * run (context, heroku) {
-  let org = context.org
+  let orgInfo = yield Utils.orgInfo(context, heroku)
+  let groupName = context.org || context.flags.team
+  let teamInviteFeatureEnabled = false
+  let isInvitedUser = false
   let email = context.args.email
 
-  let request = heroku.request({
-    method: 'DELETE',
-    path: `/organizations/${org}/members/${encodeURIComponent(email)}`
-  })
-  yield cli.action(`Removing ${cli.color.cyan(email)} from ${cli.color.magenta(org)}`, request)
+  let teamInvites = function * () {
+    return heroku.request({
+      headers: {
+        Accept: 'application/vnd.heroku+json; version=3.team-invitations'
+      },
+      method: 'GET',
+      path: `/organizations/${groupName}/invitations`
+    })
+  }
+
+  let revokeInvite = function * () {
+    let request = heroku.request({
+      headers: {
+        Accept: 'application/vnd.heroku+json; version=3.team-invitations'
+      },
+      method: 'DELETE',
+      path: `/organizations/${groupName}/invitations/${email}`
+    })
+    yield cli.action(`Revoking invite for ${cli.color.cyan(email)} in ${cli.color.magenta(groupName)}`, request)
+  }
+
+  let removeUserMembership = function * () {
+    let request = heroku.delete(`/organizations/${groupName}/members/${encodeURIComponent(email)}`)
+    yield cli.action(`Removing ${cli.color.cyan(email)} from ${cli.color.magenta(groupName)}`, request)
+  }
+
+  if (orgInfo.type === 'team') {
+    let orgFeatures = yield heroku.get(`/organizations/${groupName}/features`)
+    teamInviteFeatureEnabled = !!orgFeatures.filter(f => f.name === 'team-invite-acceptance').length
+
+    if (teamInviteFeatureEnabled) {
+      let invites = yield teamInvites()
+      isInvitedUser = !!invites.filter(m => m.user.email === email).length
```

Comment:
Same here. `filter(...).length` is a weird way to check if something exists. A couple suggestions:

Array -> map -> includes:

```javascript
> [{name: "1"},{name:"2"}].map(i => i.name).includes("2")
true
```

Array -> find

```javascript
> [{name: "1"},{name:"2"}].find(i => i.name === "2")
{ name: '2' }
```